### DISCUSSION TITLE
Problems running LFRic Apps on monsoon - export source fails

### DISCUSSION THREAD

**Discussion by stephenpcook** [MetOffice/simulation-systems#275] [Monsoon]
Running LFRic Apps on monsoon fails at the `export-source` task unless I change the cylc template to use module `scitools/production-os45-1` instead of the default `scitools`.

## Steps to reproduce

On xcsc:

``` sh
fcm co fcm:lfric_apps.x/trunk@vn1.2 lfric_apps_vn1_2
export CYLC_VERSION=8
cd lfric_apps_vn1_2/
rose stem -g gungho_model_exo-dhj-C24_MG_xc40_intel_fast-debug-64bit
cylc validate lfric_apps_vn1_2
cylc play lfric_apps_vn1_2
```

The cylc TUI reveals that the `export-source` tasks fails with the message (in `cylc-run/lfric_apps_vn1_2/runN/log/job/1/export-source/01/job.err`)

``` none
  File "/common/scitools/environments/default/2021_03_18-1/lib/python3.6/subprocess.py", line 403, in run
    with Popen(*popenargs, **kwargs) as process:
TypeError: __init__() got an unexpected keyword argument 'capture_output'
[FAIL] export_source.py # return-code=1
2024-11-20T16:02:45Z CRITICAL - failed/ERR
```

This task failing doesn't stop the cylc play, but no extra tasks are kicked off from this point.

My `~/.metomi/rose.conf` includes the automatic-options  `SITE=meto` and `PLATFORM=meto-xc40-intel`.

## Investigation

It seems that this is caused because the default scitools module has python 3.6 but the `capture_output` argument was introduced in python 3.7.

I can solve this by modifying the `METO_ORIG` cylc template, as per [lfric_apps@4907](https://code.metoffice.gov.uk/trac/lfric_apps/changeset/4907/main) (the line remove in the commit from `suite_common_xc40.cylc` was from a previous attempt to get it working).
The `export-source` then runs to completion.

**Comment by stephenpcook** [comment]
This seems to be an issue running lfric_apps generally on monsoon:

With the following commands on xcsc:
``` sh
fcm co fcm:lfric_apps.x/trunk@vn2.0 lfric_apps_vn2_0
cd lfric_apps_vn2_0/
export CYLC_VERSION=8
rose stem -g gungho_developer
cylc play lfric_apps_vn2_0
```

If I manually trigger the `export source` command from Cylc TUI, it falls down when trying to use Python 3.6.

**Reply by james-bruten-mo** [reply]
Hi Stephen,

Yes, sorry, we noticed someone else with that problem recently. Their fix is available at [this line](https://code.metoffice.gov.uk/trac/lfric_apps/changeset?reponame=&new=6455%40main%2Fbranches%2Fdev%2Fdenissergeev%2Fvn2.0.1_add_internal_flux&old=5838%40main%2Ftrunk#file4)

Feel free to copy that solution to get something working - we'll commit that change as part of the related ticket  so trunk will work once it's on.

Cheers,
James

**Reply by dennissergeev** [reply]
To give credit where it's due, it was actually @stephenpcook's solution that I copied to my branch 🙂

**Reply by james-bruten-mo** [reply]
Ah thanks Denis - I've just noticed Stephens link in his original post! Hopefully this should be sorted before too long

**Reply by stephenpcook** [reply]
Thanks @james-bruten-mo - I wasn't sure if fixing the scitools version for xc40 was the best solution but I guess that should come up in the ticket's review.
